packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -453,6 +453,21 @@ class TimelineEvent {
   bool get isCpuEventFlow => _hasChild('Engine::BeginFrame');
   bool get isGpuEventFlow => _hasChild('PipelineConsume');
 
+  /// Returns the depth of this TimelineEvent tree, including [this].
+  int get depth => _depth ?? _getDepth(this);
+  int _depth;
+
+  int _getDepth(TimelineEvent root) {
```

Comment:
this should be an instance method instead of taking a timeline event as a parameter.
Also, any reason to not return _depth at the start of the method if it is not null?
That way you could just make this a depth getter that calls the depth getter for all children to recompute the depth.

Reply:
Ohhh okay I didn't understand what you were saying before. Fixed.